cmd/podman/images/scp.go

```diff
@@ -0,0 +1,350 @@
+func makeRemoteFile(dial *ssh.Client) (string, error) {
+	run := "mktemp"
+	remoteFile, err := connection.ExecRemoteCommand(dial, run)
+	remoteFile = strings.TrimSuffix(remoteFile, "\n")
+	if err != nil {
+		return "", err
+	}
+	return remoteFile, nil
+}
+
+// createConnections takes a list of ssh urls and identity keys, and configures ssh client
+// configs for each specified host. it then dials these and creates the proper remote files for saving/loading.
+func createConnections(uri []*urlP.URL, iden []string) ([]*ssh.Client, []string, error) {
+	connList := []*ssh.Client{}
+	remoteFiles := []string{}
+	for i := range uri {
+		cfg, err := connection.ValidateAndConfigure(uri[i], iden[i])
+		if err != nil {
+			return nil, nil, err
+		}
+		dialAdd, err := ssh.Dial("tcp", uri[0].Host, cfg) // dial the client
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "failed to connect")
+		}
+		connList = append(connList, dialAdd)
+		file, err := makeRemoteFile(dialAdd)
+		if err != nil {
+			return nil, nil, err
+		}
+		remoteFiles = append(remoteFiles, file)
+	}
+	return connList, remoteFiles, nil
+}
+
+// parseLocation returns the valid connection data based off of the information provided by the user
+// iden is a string array of ssh identity keys, url is a string array of paths to ssh clients. The first entry
+// in each array is the save location and the second entry is the load location if applicable. An empty string will be
+// inserted if identity is blank (not required in some cases).
+// serv is a map of service destinations with the connection name as the index intended to be used as EngineConfig.ServiceDestinations
+// this function modifies the global scpOpt entities: FromRemote, ToRemote, Connections, and DestImageName
+func parseArgs(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination, error) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	cliConnections := []string{}
+	switch len(args) {
+	case 1:
+		if strings.Contains(args[0], "::") {
+			scpOpts.FromRemote = true
+			cliConnections = append(cliConnections, args[0])
+		} else {
+			scpOpts.DestImageName = args[0]
+			named, err := reference.ParseNormalizedNamed(scpOpts.DestImageName)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			scpOpts.DestImageName = named.String()
+		}
+	case 2:
+		if strings.Contains(args[0], "::") {
+			if !(strings.Contains(args[1], "::")) && len(strings.Split(args[0], "::")) != 2 { // if an image is specified, this mean we are loading from our client
+				cliConnections = append(cliConnections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")) == 1 || (len(strings.Split(args[0], "::")) == 2 && len(strings.Split(args[1], "::")) == 2) { // is save->load w/ one image name or save-load w/ two image names
```

Comment:
From https://github.com/containers/podman/pull/10828#discussion_r673317950 :

If `strings.Contains(args[0], "::")`, as is enforced above above, `len(strings.Split(args[0], "::"))` is always ≥ 2. Same for `args[1]`, so this is AFAICS equivalent to `if true`.

Reply:
resolved above